source/objects/terrain.lua

```diff
@@ -132,32 +132,13 @@ function Terrain.generate(intAmountToCreate)
 	groundTableSize = #GROUND
 
 	-- add some buildings before adding fuel
-
-	repeat
-		local lastBuildingIndex
-		local nextBuildingX
-
-		-- get the index/id of the last building
-		lastBuildingIndex = getLastBaseID(enum.basetypeBuilding)
-
-		if lastBuildingIndex < 1 then
-			nextBuildingX = gintOriginX + love.math.random((gintScreenWidth / 2),gintScreenWidth)
-		else
-			-- the next building is between one screenwidth and 1.66 away from the last building
-			local nextBuildingDistance = gintScreenWidth + love.math.random((gintScreenWidth * 0.66),gintScreenWidth)
-			nextBuildingX = garrObjects[lastBuildingIndex].x + nextBuildingDistance
-		end
-		nextBuildingX = cf.round(nextBuildingX,0)
-		if nextBuildingX <= groundTableSize then
-			local newBaseType = love.math.random(7,8)		-- hack
-			cobjs.CreateObject(newBaseType, nextBuildingX)
-		else
-			break
-		end
-	until not true	-- infinite loop using a break statement
-
-	-- add fuel bases after the buildings so they can draw layered if need be
-	addFuelBases(groundTableSize)
+	if LANDERS[1] == nil then
+	
+	else
+		addBuildings(LANDERS[1].x + 6000)	-- an arbitrary 'draw ahead' distance
+		-- add fuel bases after the buildings so they can draw layered if need be
+		addFuelBases(LANDERS[1].x + 6000)	-- an arbitrary 'draw ahead' distance
+	end
```

Comment:
The lander needs to have at least one fuel base generated before the lander itself is created because `checkForContact(lander, dt)` will call `Fun.GetDistanceToClosestBase` as soon as the lander spawns. 

I'll try to quickly work around that for the build to not always crash when trying to load a game.

Quite possible that we have to push the v0.11 release out a bit to get that sorted out.

Reply:
Okay, the issue causing the crash actually was [a loading problem](https://github.com/togfoxy/MarsLander/pull/232#pullrequestreview-802844247).

But still, this behavior described here is something that should be fixed since the lander always needs at least one fuel base to not crash the game. We need to add a check so that the game is playable even if there's no fuel base generated.

In general it is desirable to have all objects working without dependencies like this one.